### PR TITLE
APIMF-3247: removed media-types and side-fixes

### PR DESCRIPTION
**Pull request by tomsfernandez**

### CODE REVIEW

File: shared/src/main/scala/amf/core/client/common/validation/ProfileNames.scala [hunk 1]
```diff
@@ -68,22 +68,22 @@ object ProfileName {
...
 
   def apply(profile: String): ProfileName = profile match {
-    case Amf.name             => AmfProfile
-    case "OAS" | Oas20.name   => Oas20Profile // for compatibility
-    case Oas30.name           => Oas30Profile
-    case Raml08.name          => Raml08Profile
-    case "RAML" | Raml10.name => Raml10Profile // for compatibility
-    case AsyncApi.name        => AsyncProfile
-    case AsyncApi20.name      => Async20Profile
+    case Amf.`id`             => AmfProfile
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/common/validation/ProfileNames.scala]: Remove backticks

File: shared/src/main/scala/amf/core/client/platform/model/document/BaseUnit.scala [hunk 2]
```diff
@@ -66,8 +66,7 @@ trait BaseUnit extends AmfObjectWrapper with PlatformSecrets {
   def findByType(typeId: String): ClientList[DomainElement] =
     _internal.findByType(Namespace.defaultAliases.expand(typeId).iri()).asClient
 
-  def sourceMediaType: String            = _internal.sourceMediaType
-  def sourceVendor: ClientOption[Vendor] = _internal.sourceVendor.asClient
+  def sourceVendor: ClientOption[Spec] = _internal.sourceSpec.asClient
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/platform/model/document/BaseUnit.scala]: Rename to sourceSpec

File: shared/src/main/scala/amf/core/client/scala/AMFGraphConfiguration.scala [hunk 3]
```diff
@@ -139,7 +141,7 @@ class AMFGraphConfiguration private[amf] (override private[amf] val resolvers: A
     * Merges two environments taking into account specific attributes that can be merged.
     * This is currently limited to: registry plugins, registry transformation pipelines.
     */
-  def merge(other: AMFGraphConfiguration): AMFGraphConfiguration = super._merge(other)
+  protected[amf] def merge(other: AMFGraphConfiguration): AMFGraphConfiguration = super._merge(other)
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/scala/AMFGraphConfiguration.scala]: Remove, doesn't make sense now.

File: shared/src/main/scala/amf/core/internal/annotations/serializable/CoreSerializableAnnotations.scala [hunk 4]
```diff
@@ -9,7 +9,7 @@ private[amf] object CoreSerializableAnnotations extends SerializableAnnotations
       "lexical"              -> LexicalInformation,
       "host-lexical"         -> HostLexicalInformation,
       "base-path-lexical"    -> BasePathLexicalInformation,
-      "source-vendor"        -> SourceVendor,
+      "source-vendor"        -> SourceSpec,
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/internal/annotations/serializable/CoreSerializableAnnotations.scala]: Add "todo" to change this

File: shared/src/main/scala/amf/core/internal/plugins/parse/DomainParsingFallback.scala [hunk 5]
```diff
@@ -1,30 +1,59 @@
...
-          .withEncodes(
-              ExternalDomainElement()
-                .withRaw(root.raw)
-                .withMediaType(root.mediatype))
-    }
+case class ExternalFragmentDomainFallback(strict: Boolean = true) extends DomainParsingFallback {
+  override def chooseFallback(root: Root, availablePlugins: Seq[AMFParsePlugin], isRoot: Boolean): AMFParsePlugin = {
+    if (isRoot && strict) throw UnsupportedDomainForDocumentException(root.location)
+    else {
+      new AMFParsePlugin {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/internal/plugins/parse/DomainParsingFallback.scala]: Extract

File: shared/src/main/scala/amf/core/internal/transform/pipelines/BasicTransformationPipeline.scala [hunk 6]
```diff
@@ -13,12 +13,12 @@ class BasicTransformationPipeline private (override val name: String) extends Tr
 }
 
 object BasicTransformationPipeline {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/internal/transform/pipelines/BasicTransformationPipeline.scala:16]: Does this pipeline make sense??

File: shared/src/main/scala/amf/core/client/scala/AMFResult.scala [hunk 7]
```diff
@@ -13,4 +14,9 @@ import amf.core.client.scala.validation.{AMFValidationResult, ReportConformance}
   */
 case class AMFResult(baseUnit: BaseUnit, results: Seq[AMFValidationResult]) extends AMFObjectResult(baseUnit, results)
 
+class AMFParseResult(override val baseUnit: BaseUnit, override val results: Seq[AMFValidationResult])
+    extends AMFResult(baseUnit, results) {
+  def rootSpec: Spec = baseUnit.sourceSpec.getOrElse(Amf)
```
Comment (AgustinBettati) [inline shared/src/main/scala/amf/core/client/scala/AMFResult.scala:19]: Will we expose rootSpec in AMFResult and also in BaseUnit?

Comment (tomsfernandez) [reply shared/src/main/scala/amf/core/client/scala/AMFResult.scala:19]: I will leave it like that for the moment.

**Pull request merged**